torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2269,6 +2275,23 @@ def fn(op_info, device, dtype, requires_grad):
 
     return fn
 
+def sample_inputs_nan_reduction(supports_multiple_dims):
```

Comment:
Seems like this should either combine the above and the normal inputs (to verify it handles non-nan inputs properly) or the above nan inputs should be part of the regular inputs to every reduction operator and the nan ops shouldn't have a special-cased sample inputs